### DISCUSSION TITLE
Support for .svelte files?

### DISCUSSION THREAD

**Discussion by kamiyaa** [biomejs/biome#136] [Ideas]
Any chance biome will be able to format `.svelte` files as well?
Or other non css/js/ts/json files?

**Comment by ematipico** [comment]
Biome can lint and format JSON/JSONC files. 

We're now working towards CSS support.

Next will be HTML/markdown. With HTML supported, we might be able to support svelte too. It's a long way to go though

**Reply by enBonnet** [reply]
That would be wonderful.

**Reply by KyTiXo** [reply]
Kudos! Very much wanted + Svelte :D

**Reply by glitch-txs** [reply]
looking forward for Svelte support <3

**Reply by orbiteleven** [reply]
Is there a way we can track the progress of this?

**Reply by ematipico** [reply]
> Is there a way we can track the progress of this? 

https://biomejs.dev/internals/language-support/

**Reply by orbiteleven** [reply]
I guess I mean something more "push" than "pull". If there were a tracking issue in Github to which one could subscribe, it would be more likely that I/we could engage early when things are ready.

Looking forward to it.

**Reply by ematipico** [reply]
Usually we don't create issues unless we start the work, otherwise that would send incorrect signals to people.

We will start once people want and can work on that. If there's someone that wants to help kicking off the works, that's more than welcome!

**Reply by orbiteleven** [reply]
That makes sense, and also tells me it's not underway now. Fingers crossed that someone with time and experience gets to it soon. You don't want me writing Rust :D

**Comment by hankadler** [comment]
Simply treating whatever is inside `<script> ... </script>` as JavaScript and `<script lang="ts"> ... </script>` as TypeScript so that the same fixes that apply to a `.js/.ts` file applied to code inside script block in a `.svelte` file would go a long way. The only syntactical difference there is reactivity; statements beginning with `$:`, everything else is good old js/ts.

**Reply by nhedger** [reply]
It was my understanding that `$:` is valid JS. If that’s the case we could try adding initial support based on what we just did for Vue SFC.

There’s actually a good chance we can reuse the same logic to parse the script block.

If you’re interested in contributing I encourage you to check the following resources:

https://github.com/biomejs/biome/issues/1719

https://github.com/biomejs/biome/pull/1722

**Reply by ematipico** [reply]
It's valid JavaScript, it's a label. For the Svelte compiler, it holds different semantics at compile time. We don't care about such semantics for now.

**Comment by isAlmogK** [comment]
I would also like this, Following to be updated would be great if there was an open issue just for Svelte support

**Reply by enBonnet** [reply]
They are working on it https://biomejs.dev/internals/language-support/

**Reply by isAlmogK** [reply]
Thanks

**Comment by Netail** [comment]
Hey!
As of [Biome 2.3](https://biomejs.dev/blog/biome-v2-3/) we support HTML, Astro, Svelte & Vue under the experimental option: `html.experimentalFullSupportEnabled`.
Support is still experimental due to potentially missed edge cases, but definitely worth trying :) 
In case you find any issues please make sure to [create a GitHub issue](https://github.com/biomejs/biome/issues/new/choose) so we can get closer to stable :)

More on HTML-ish languages can be found here: https://biomejs.dev/internals/language-support/#html-super-languages-support
